### PR TITLE
[doc] Split out docs pages for rv_dm, otbn, and rv_core_ibex

### PR DESCRIPTION
**Pull request by jwnrt**
We split out "theory of operation" and "programmer's guide" sections of each IP block's docs into separate pages in https://github.com/lowRISC/opentitan/pull/17467.

This PR splits out the `rv_dm` docs which might have been missed, as well as the Ibex and OTBN core documentation.

* Fixed a Tabel -> Table typo.
* Split out rv_dm's theory of operation and programmer's guide
* Split out rv_core_ibex's theory of operation and programmer's guide
* Split ouf otbn's theory of operation and programmer's guide
  * Renamed the "Developer Guide" page to "Developing OTBN" to avoid confusion with the "Programmer's Guide"

Updated the relative links on these sub-pages now that the content has moved.

### CODE REVIEW

**Comment by jwnrt** [comment]
Sorry, didn't run the link checker before pushing. Fixed those broken links.

File: hw/ip/otp_ctrl/doc/programmers_guide.md
Comment (marnovandermaas) [inline hw/ip/otp_ctrl/doc/programmers_guide.md:227]: Nice catch!

**Review by marnovandermaas — APPROVED** [review]
LGTM, I'll wait for one more approver before merging.

**Pull request merged**
